polaris.shopify.com/src/components/ComponentExamples/ComponentExamples.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useState } from "react";
+import styles from "./ComponentExamples.module.scss";
+import CodesandboxButton from "../CodesandboxButton";
+import Code from "../Code";
+import { Tab } from "@headlessui/react";
+import { className } from "../../utils/various";
+
+export type ComponentExample = {
+  code: string;
+  description: string;
+  fileName: string;
+  title: string;
+};
+
+interface Props {
+  examples: ComponentExample[];
+}
+
+// https://stackoverflow.com/a/60338028
+function formatHTML(html: string): string {
+  const tab = "  ";
+  let result = "";
+  let indent = "";
+
+  html.split(/>\s*</).forEach((element) => {
+    if (element.match(/^\/\w/)) {
+      indent = indent.substring(tab.length);
+    }
+    result += indent + "<" + element + ">\r\n";
+
+    if (element.match(/^<?\w[^>]*[^\/]$/) && !element.startsWith("input")) {
+      indent += tab;
+    }
+  });
+
+  return result.substring(1, result.length - 3);
+}
+
+const ComponentExamples = (props: Props) => {
+  const { examples } = props;
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [htmlCode, setHTMLCode] = useState("");
+
+  const [iframeHeight, setIframeHeight] = useState("400px");
+
+  const handleExampleLoad = () => {
+    let attempts = 0;
+
+    const waitForExampleContent = setInterval(() => {
+      const exampleIframe = document.getElementById(
+        "examples-iframe"
+      ) as HTMLIFrameElement;
+      const exampleElement = exampleIframe?.contentDocument;
+      const exampleWrapper = exampleElement?.getElementById("polaris-example");
+
+      const padding = 192;
+      let newHeight = padding;
+
+      if (exampleWrapper) {
+        newHeight += exampleWrapper?.offsetHeight;
+        setIframeHeight(`${newHeight}px`);
+        setHTMLCode(formatHTML(exampleWrapper.innerHTML));
+        clearInterval(waitForExampleContent);
+      }
+
+      attempts++;
+
+      if (attempts > 10) {
+        clearInterval(waitForExampleContent);
+      }
+    }, 100);
+
+    return () => {
+      clearInterval(waitForExampleContent);
+    };
+  };
+
+  useEffect(() => {
+    setSelectedIndex(0);
+  }, [examples]);
+
+  return (
+    <>
+      <h2 id="examples">Examples</h2>
+
+      <Tab.Group
```

Comment:
Maaaaybe! Although I'm not sure we would want this styling for other types of tabs — it's very specific for the `Code` component. All the logic and a11y features come from `HeadlessUI` so that part is already reusable for other parts of the site.